crates/types/src/uint.rs

```diff
@@ -263,7 +263,7 @@ where
     }
 
     fn checked_mul_dec_ceil(self, rhs: F) -> StdResult<Self> {
-        if rhs.is_zero() {
+        if self.is_zero() || rhs.is_zero() {
             return Ok(Self::ZERO);
         }
```

Comment:
Math bug here. If lhs (`self`) is zero, then later in `self.checked_multiply_ratio_ceil` we get a division by zero error.